src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java

```diff
@@ -413,15 +399,15 @@ public void put(Collection<SinkRecord> records) throws ConnectException {
     ) {
         Map<TopicPartition, OffsetAndMetadata> offsetsToCommit = new HashMap<>();
         for (TopicPartition tp : assignment) {
-            if(writers.get(tp) == null) {
+            if (writers.get(tp) == null) {
                 throw new ConnectException("Topic Partition not configured properly. " +
                         "verify your `topics` and `kusto.tables.topics.mapping` configurations");
             }
 
             Long lastCommittedOffset = writers.get(tp).lastCommittedOffset;
 
             if (lastCommittedOffset != null) {
-                Long offset = lastCommittedOffset + 1L;
+                long offset = lastCommittedOffset + 1L;
```

Comment:
please revert as lastCommittedOffset itself is Long

Reply:
lastCommittedOffset must be an Object because it might be null. But "offset" can be autounboxed to a primitive, which is preferable for performance reasons when an Object isn't needed. In this case, a primitive suffices. It won't even have to be autoboxed into a Long when creating a new OffsetAndMetadata, as that accepts a primitive.

Reply:
then maybe just change the 1L to 1l ?

Reply:
Both lower-case and upper-case "L" are identical in Java, and both represent a long (primitive). When the object being assigned to (left-hand side) was Long (object), the primitive was being autoboxed behind the scenes.